core/src/main/java/com/rudderstack/android/sdk/core/RudderDataResidencyManager.java

```diff
@@ -64,70 +68,66 @@ void processDataPlaneUrl() {
-        if (dataResidencyUrl != null) {
-            setDataPlaneUrl(dataResidencyUrl);
-        }
+    @Nullable
+    private String handleDefaultServer() {
+        return getDataResidencyUrl(RudderDataResidencyServer.US.name());
     }
 
     /**
      * Look for the region in the data residency list.
      *
-     * @param region Data Residency region set while SDK initialisation.
-     * @return Corresponding residency URL if present else null.
+     * @param region Data Residency region which is set while SDK initialisation.
+     * @return Corresponding residency URL if present otherwise null.
      */
     @VisibleForTesting
     @Nullable
-    String getDataResidencyUrl(String region) {
-        if (isEmpty(dataResidencyUrls)) {
+    String getDataResidencyUrl(@NonNull String region) {
+        if (isEmpty(dataResidencyUrls) ||
+                !dataResidencyUrls.containsKey(region) || isEmpty(dataResidencyUrls.get(region))) {
             return null;
         }
-        region = region.toLowerCase(Locale.ROOT);
-        if (dataResidencyUrls.containsKey(region)) {
-            String dataResidencyUrl = dataResidencyUrls.get(region);
-            if (!isEmpty(dataResidencyUrl)) {
-                return appendSlashToUrl(dataResidencyUrl);
+        for (Map<String, Object> dataResidencyUrl : dataResidencyUrls.get(region)) {
+            ResidencyUrl residencyUrl = getResidencyUrl(dataResidencyUrl);
+            if (residencyUrl.defaultTo) {
+                if (!isEmpty(residencyUrl.url)) {
+                    return appendSlashToUrl(residencyUrl.url);
+                }
+                // It is decided that for any region there will be only one url defaulted to true
+                break;
             }
         }
         return null;
     }
 
-    // Getter and Setter
-
-    /**
-     *
-     * @return dataPlaneUrl from the sourceConfig based on the ResidencyServer region.
-     *
-     * If either required region is not present in the sourceConfig or sourceConfig is null
-     * then return null.
-     */
-    @Nullable
-    String getDataPlaneUrl() {
-        return this.dataPlaneUrl;
+    private static class ResidencyUrl {
+        String url = null;
+        boolean defaultTo = false;
     }
 
-    @VisibleForTesting
-    void setDataPlaneUrl(String dataPlaneUrl) {
-        this.dataPlaneUrl = dataPlaneUrl;
+    @NonNull
+    private ResidencyUrl getResidencyUrl(Map<String, Object> dataResidencyUrl) {
+        ResidencyUrl residencyUrl = new ResidencyUrl();
+        if (isEmpty(dataResidencyUrl)) {
+            return residencyUrl;
+        }
+        if (dataResidencyUrl.containsKey("url")) {
+            residencyUrl.url = getString(dataResidencyUrl.get("url"));
+        }
+        if (dataResidencyUrl.containsKey("default")) {
+            residencyUrl.defaultTo = getBoolean(dataResidencyUrl.get("default"));
+        }
+        return residencyUrl;
     }
```

Comment:
Instead of defining the data class `ResidencyUrl` here and manually converting the generic object to Residency URL, 
can we define this in the `RudderServerConfigSource` data class itself and let the gson do the conversion automatically for us ?

Reply:
I tried that approach initially but because of some constraints it was not possible to achieve it using gson. Hence I proceeded with the current approach.
Contraints:

- Objects inside the `dataplanes` don't have any fixed `key` (e.g., EU or US could vary). Hence gson wouldn't work.
- Inside any particular region (e.g., say inside US) we don't have `key-value` pair, instead an `array`. Hence didn't found a way to store them using gson.

Please LMK a way, if we could improve this portion further with the help of gson. I'll try it out.